lib/Zonemaster/CLI.pm

```diff
@@ -752,23 +756,24 @@ sub add_fake_delegation {
     foreach my $pair ( @{ $self->ns } ) {
         my ( $name, $ip ) = split( '/', $pair, 2 );
 
-        if ( not $name ) {
+        if ( $pair =~ tr/\/// > 1 or not $name ) {
```

Comment:
This looks to me like a tightening of the validation of the IP part. Why not perform the full IP address validation here instead of just looking or extra slashes?

Reply:
I really like the idea of passing all inputs through some kind of dedicated validation layer close to where they enter. That way it's easy to find out *if*, *where* and *how* all inputs are validated. It's also easier to provide good user feedback. E.g. instead of saying that the "ip" method can't be called on an undefined value, we could be saying that the IP address part of the `--ns` argument is invalid.

In Backend we have one validation layer for RPCAPI requests in zm-rpcapi and another one for the config file that is shared by zm-rpcapi and zm-testagent. In CLI, on the other hand, things are not so tidy and we're mixing validation and business logic. IMHO we should refactor CLI to have a proper validation layer too. In fact I'll create and issue for that.

Regarding this PR though, if you think it's out of scope for this PR to improve old error messages I'm perfectly fine with that. I think it's a great improvement that you added the validation for extra slashes.